IPython/core/magics/code.py

```diff
@@ -258,25 +258,39 @@ def pastebin(self, parameter_s=''):
 
           -d: Pass a custom description. The default will say
               "Pasted from IPython".
+          -e: Pass number of days for the link to be expired.
+              The default will be 7 days.
         """
-        opts, args = self.parse_options(parameter_s, 'd:')
+        opts, args = self.parse_options(parameter_s, "d:e:")
 
         try:
             code = self.shell.find_user_code(args)
         except (ValueError, TypeError) as e:
             print(e.args[0])
             return
 
+        expiry_days = 7
```

Comment:
Note to others: this is the same value as the current dpaste's default, so setting it preemptively does not change current behavior.